File: backend/internal/model/quest.go
package model

// QuestSession holds the in-memory state for an active quest.
type QuestSession struct {
	PokemonID     int     `json:"pokemon_id"`
	DescriptionEN string  `json:"description_en"`
	DescriptionJA string  `json:"-"`
	NameEN        string  `json:"-"`
	NameJA        string  `json:"-"`
	SpriteURL     string  `json:"-"`
	BaseStatTotal int      `json:"-"`
	Types         []string `json:"-"`
	Height        int      `json:"-"`
	Weight        int      `json:"-"`
	IsLegendary   bool     `json:"-"`
	IsMythical    bool     `json:"-"`
	Score         float64  `json:"score,omitempty"`
	BallType      string  `json:"ball_type,omitempty"`
	GuessAttempts int     `json:"guess_attempts,omitempty"`
	NameGuessed   bool    `json:"-"`
}

// ScoreResult is the AI-agnostic result of a translation scoring.
type ScoreResult struct {
	Score   float64 `json:"score"`
	Review  string  `json:"review"`
}

// ChatMessage represents a single message in the professor chat.
type ChatMessage struct {
	Role    string `json:"role"`    // "user" or "professor"
	Content string `json:"content"`
}

// ChatContext holds the quest context needed for the professor chat.
type ChatContext struct {
	DescriptionEN string  `json:"description_en"`
	DescriptionJA string  `json:"description_ja"`
	Translation   string  `json:"translation"`
	Score         float64 `json:"score"`
	Review        string  `json:"review"`
	NameEN        string  `json:"name_en"`
	NameJA        string  `json:"name_ja"`
}

File: backend/src/types/quest.ts
/** 進行中のクエストセッション。出題から捕獲までの状態を保持する。 */
export interface QuestSession {
  pokemon_id: number;
  description_en: string;
  description_ja: string;
  name_en: string;
  name_ja: string;
  sprite_url: string;
  base_stat_total: number;
  types: string[];
  height: number;
  weight: number;
  is_legendary: boolean;
  is_mythical: boolean;
  score: number;
  ball_type: string;
  guess_attempts: number;
  name_guessed: boolean;
}

/** AIScorer.scoreTranslation の戻り値。 */
export interface ScoreResult {
  score: number;
  review: string;
}

/** チャットの 1 発言。 */
export interface ChatMessage {
  role: "user" | "professor";
  content: string;
}

/** オーキド博士チャットに与えるクエスト文脈。 */
export interface ChatContext {
  description_en: string;
  description_ja: string;
  translation: string;
  score: number;
  review: string;
  name_en: string;
  name_ja: string;
}
